pkg/compliance/checks/kubeapiserver_check_test.go

```diff
@@ -14,11 +14,98 @@ import (
 
 	assert "github.com/stretchr/testify/require"
 
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/client-go/dynamic/fake"
+	kscheme "k8s.io/client-go/kubernetes/scheme"
 )
 
+var scheme = kscheme.Scheme
+
+type MyObj struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec MyObjSpec `json:"spec,omitempty"`
+}
+type MyObjSpec struct {
+	StringAttribute string                 `json:"stringAttribute,omitempty"`
+	BoolAttribute   bool                   `json:"boolAttribute,omitempty"`
+	ListAttribute   []interface{}          `json:"listAttribute,omitempty"`
+	StructAttribute map[string]interface{} `json:"structAttribute,omitempty"`
+}
+
+type MyObjList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []MyObj
+}
+
+func (in *MyObj) DeepCopyInto(out *MyObj) {
+	*out = *in
+	out.TypeMeta = in.TypeMeta
+	in.ObjectMeta.DeepCopyInto(&out.ObjectMeta)
+}
+
+func (in *MyObj) DeepCopy() *MyObj {
+	if in == nil {
+		return nil
+	}
+	out := new(MyObj)
+	in.DeepCopyInto(out)
+	return out
+}
+
+func (in *MyObj) DeepCopyObject() runtime.Object {
+	if c := in.DeepCopy(); c != nil {
+		return c
+	}
+	return nil
+}
+
+func (in *MyObjList) DeepCopy() *MyObjList {
+	if in == nil {
+		return nil
+	}
+	out := new(MyObjList)
+	in.DeepCopyInto(out)
+	return out
+}
+
+func (in *MyObjList) DeepCopyInto(out *MyObjList) {
+	*out = *in
+	out.TypeMeta = in.TypeMeta
+	in.ListMeta.DeepCopyInto(&out.ListMeta)
+	if in.Items != nil {
+		in, out := &in.Items, &out.Items
+		*out = make([]MyObj, len(*in))
+		for i := range *in {
+			(*in)[i].DeepCopyInto(&(*out)[i])
+		}
+	}
+}
+
+func (in *MyObjList) DeepCopyObject() runtime.Object {
+	if c := in.DeepCopy(); c != nil {
+		return c
+	}
+	return nil
+}
+
+func addKnownTypes(scheme *runtime.Scheme) error {
+	scheme.AddKnownTypes(schema.GroupVersion{Group: "mygroup.com", Version: "v1"},
+		&MyObj{},
+		&MyObjList{},
+	)
+	return nil
+}
+
+func init() {
+	schemeBuilder := runtime.NewSchemeBuilder(addKnownTypes)
+	schemeBuilder.AddToScheme(scheme)
+}
```

Comment:
you could move this function on top of the file

Reply:
:+1: Done: https://github.com/DataDog/datadog-agent/pull/7857/commits/067fca31ab65d678c7b2056c1a3438ed22b16113